### ISSUE TITLE
In ctsm python library: get more machine defaults from config_machines.xml

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/ctsm#494]
This issue refers to code that is not yet on master: #493 .

Currently, I have started a machine_defaults.py that gives defaults for some parameters that we need for supported machines. Long-term it would be good to get as much as possible from config_machines.xml, via cime functions. I just haven't taken the time to figure out how to do this right now. This would avoid duplication, reduce maintenance, and allow our python stuff to run on other supported machines without our needing to introduce our own defaults for (e.g.) the scratch directory on each machine.

Specifically: Try to (a) determine machine name based on the existing nodename_regex: is there an easy-to-use cime function for that?; (b) use CIME_OUTPUT_ROOT rather than having our own defaults for the scratch directory on a machine; (c) see if we could also replace other aspects of our machine_defaults.py with stuff from cime (if not, consider whether we'd want to move some of this stuff into config_machines eventually).

**Comment by ekluzek** [comment]
@billsacks something else I need to do is to get the list of resolutions from config_grids.xml. I'm not sure if it's best to use query_config to do that, or if using Python objects is the way to go. But, I want to throw this out there as a need in case, any of what you are doing overlaps with that need as well.

**Comment by billsacks** [comment]
@ekluzek Indeed, part of my hope in putting some of this together is that it might pave the way to using some more pieces of cime library directly from python scripts in CTSM, rather than calling scripts and parsing their output. That said, I have no idea what it will take to get the grid information you need from a CIMElib function.
